source/extensions/tracers/xray/tracer.cc

```diff
@@ -89,31 +89,35 @@ void Span::finishSpan() {
     s.mutable_http()->mutable_response()->insert(KeyValue{item.first, item.second});
   }
 
-  // TODO(marcomagdy): test how expensive this validation is. Might be worth turning off in
-  // optimized builds..
-  MessageUtil::validate(s, ProtobufMessage::getStrictValidationVisitor());
-  Protobuf::util::JsonPrintOptions json_options;
-  json_options.preserve_proto_field_names = true;
-  std::string json;
-  const auto status = Protobuf::util::MessageToJsonString(s, &json, json_options);
-  ASSERT(status.ok());
+#ifndef NDEBUG
```

Comment:
Would prefer to not have this type of conditional here. I would either include this code or not. If there is some good reason to do this move this check into an ASSERT and make sure you have tests for the validation path.

Reply:
Yeah I was between removing it and wrapping it with ifdef. So, I'll just remove it.